ddl/ddl_api.go

```diff
@@ -1073,6 +1073,25 @@ func getDefaultValue(ctx sessionctx.Context, col *table.Column, c *ast.ColumnOpt
 
 		return value, false, nil
 	}
+
+	switch x := c.Expr.(type) {
+	case *ast.FuncCallExpr:
+		if x.FnName.L == ast.Rand {
+			if err := expression.VerifyArgsWrapper(ast.Rand, len(x.Args)); err != nil {
+				return nil, false, expression.ErrIncorrectParameterCount.GenWithStackByArgs(ast.Rand)
+			}
+			col.DefaultIsExpr = true
+			var sb strings.Builder
+			restoreFlags := format.RestoreStringSingleQuotes | format.RestoreKeyWordLowercase | format.RestoreNameBackQuotes |
+				format.RestoreSpacesAroundBinaryOperation
+			restoreCtx := format.NewRestoreCtx(restoreFlags, &sb)
+			if err := c.Expr.Restore(restoreCtx); err != nil {
+				return "", false, err
+			}
+			return sb.String(), false, nil
```

Comment:
`rand()` function returns a random floating-point value, but here you return a string, seems it is not what the user wants

Reply:
It is a mark for later inserting values into rows, which would be overwritten by calculating the expression `rand()`  in `func EvalColDefaultExpr`.